distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Cast_Helpers.enso

```diff
@@ -0,0 +1,56 @@
+from Standard.Base import all
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+
+import project.Data.Type.Value_Type.Value_Type
+import project.Internal.Parse_Values_Helper
+from project.Errors import Lossy_Conversion
+
+polyglot java import org.enso.table.data.column.operation.CastProblemBuilder
+
+## PRIVATE
+   Checks if one type can be cast into another and returns a dataflow error
+   explaining the situation if not.
+check_cast_compatibility source_type target_type =
+    if (target_type == Value_Type.Mixed) || target_type.is_text || (source_type == target_type) then True else
+        if source_type.is_text && is_a_valid_parse_target target_type then Error.throw (Illegal_Argument.Error "To parse a text column into "+target_type.to_display_text+" type, `parse` should be used instead of `cast`.") else
+            if target_type.is_numeric then source_type.is_numeric else
+                if source_type == Value_Type.Boolean then target_type.is_numeric else
+                    case source_type of
+                        Value_Type.Date_Time _ ->
+                            (target_type == Value_Type.Date) || (target_type == Value_Type.Time)
+                        Value_Type.Date -> target_type.has_date
+                        Value_Type.Binary _ _ -> case target_type of
```

Comment:
No, because the type `Binary` has 2 parameters.
`target_type == Value_Type.Binary` will only work if the `target_type` is the `Value_Type.Binary` instantiated with the defaults, i.e. `Value_Type.Binary Nothing False`.
But I want this case to work with instances that have _any_ parameters - as I _do want_ to allow conversions between `Binary` types with different settings.

Reply:
Other than that - I'm not fond of this method, it feels a bit ugly but I did not figure a way to write this logic more nicely - if you have any suggestions they would be very much welcome 😅

Reply:
Oh I missed the underscores while reading the code.